tests/virtctl/memorydump.go

```diff
@@ -0,0 +1,322 @@
+package virtctl
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	k8sv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/libvmi"
+	"kubevirt.io/kubevirt/tests/clientcmd"
+	"kubevirt.io/kubevirt/tests/decorators"
+	"kubevirt.io/kubevirt/tests/exec"
+	"kubevirt.io/kubevirt/tests/framework/checks"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	. "kubevirt.io/kubevirt/tests/framework/matcher"
+	"kubevirt.io/kubevirt/tests/libstorage"
+	"kubevirt.io/kubevirt/tests/libvmifact"
+	"kubevirt.io/kubevirt/tests/testsuite"
+)
+
+var _ = Describe("[sig-storage][virtctl]Memory dump", decorators.SigStorage, func() {
+	const (
+		claimNameFlag   = "--claim-name"
+		createClaimFlag = "--create-claim"
+		outputFlag      = "--output"
+		portForwardFlag = "--port-forward"
+	)
+
+	var (
+		pvcName string
+		vm      *v1.VirtualMachine
+	)
+
+	BeforeEach(func() {
+		if _, exists := libstorage.GetRWOFileSystemStorageClass(); !exists {
+			Fail("Fail no filesystem storage class available")
+		}
+
+		pvcName = "fs-pvc-" + rand.String(5)
+
+		vm = libvmi.NewVirtualMachine(libvmifact.NewCirros(), libvmi.WithRunStrategy(v1.RunStrategyAlways))
+		var err error
+		vm, err = kubevirt.Client().VirtualMachine(testsuite.GetTestNamespace(nil)).Create(context.Background(), vm, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		Eventually(ThisVM(vm), 360*time.Second, 1*time.Second).Should(BeReady())
+	})
+
+	DescribeTable("Should be able to get and remove memory dump", func(create bool) {
+		args := []string{
+			claimNameFlag, pvcName,
+		}
+		if create {
+			args = append(args, createClaimFlag)
+		} else {
+			libstorage.CreateFSPVC(pvcName, testsuite.GetTestNamespace(nil), "500Mi", nil)
+		}
+
+		Expect(runMemoryDumpGetCmd(vm.Name, args...)).To(Succeed())
+		out := waitForMemoryDumpCompletion(vm.Name, pvcName, "", false)
+		Expect(runMemoryDumpRemoveCmd(vm.Name)).To(Succeed())
+		waitForMemoryDumpDeletion(vm.Name, pvcName, out, true)
+	},
+		Entry("[test_id:9034] when creating a PVC", true),
+		Entry("with an existing PVC", false),
+	)
+
+	It("[test_id:9035]Run multiple memory dumps", func() {
+		out := ""
+		for i := range 3 {
+			By(fmt.Sprintf("Running memory dump number: %d", i+1))
+			if i > 0 {
+				Expect(runMemoryDumpGetCmd(vm.Name)).To(Succeed())
+			} else {
+				Expect(runMemoryDumpGetCmd(vm.Name, claimNameFlag, pvcName, createClaimFlag)).To(Succeed())
```

Comment:
here you shouldnt use the createclaimflag since the point of this test was to do memory dump to the same pvc.
but maybe this test with virtctl is redundant anyways